File: integration-tests/java-selenium/java-selenium-test/src/main/java/com/gu/fronts/integration/test/page/common/Article.java
package com.gu.fronts.integration.test.page.common;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute;
import com.gu.fronts.integration.test.page.util.AbstractParentPage;

public class Article extends AbstractParentPage {
    @FindByTestAttribute(using = "article")
    private WebElement rootElement;
    @FindByTestAttribute(using = "article-headline")
    private WebElement headline;

    public Article(WebDriver webDriver) {
        super(webDriver);
    }

    @Override
    public Article isDisplayed() {
        assertExistsAndDisplayed(rootElement, headline);
        return this;
    }

    public String headlineText() {
        return headline.getText();
    }

}

File: integration-tests/scala-selenium/app/com/gu/fronts/integration/test/page/common/Article.scala
package com.gu.fronts.integration.test.page.common

import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement

import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute
import com.gu.fronts.integration.test.page.util.FrontsParentPage
import com.gu.fronts.integration.test.page.util.PageElementHelper.existsAndDisplayed

class Article(webDriver: WebDriver) extends FrontsParentPage(webDriver) {

  @FindByTestAttribute(using = "article")
  private var rootElement: WebElement = _

  @FindByTestAttribute(using = "article-headline")
  private var headline: WebElement = _

  override def isDisplayed(): Boolean = {
    existsAndDisplayed(rootElement, headline)
  }

  def headlineText(): String = headline.getText
}